py3status/modules/file_status.py

```diff
@@ -98,30 +104,33 @@ def file_status(self):
         paths = list(map(glob, self.path))
         # merge list of paths
         paths = [x for x in chain.from_iterable(paths)]
+        file_status_data['files'] = len(paths)
 
-        # fill data
+        # fill data, legacy stuff
         file_status_data['icon'] = self.icon_unavailable
         color = self.py3.COLOR_BAD
-
         if paths:
             file_status_data['icon'] = self.icon_available
             color = self.py3.COLOR_GOOD
 
         # format paths
-        if self.format_path:
+        if set(['basename', 'fullname']) & self.placeholders:
             format_path = {}
 
-            format_path_separator = self.py3.safe_format(
-                self.format_path_separator)
+            if set(['basename']) & self.placeholders:
+                format_path['basename'] = map(basename, paths)
 
-            format_path['basename'] = self.py3.composite_join(
-                format_path_separator, map(basename, paths))
+            if set(['fullpath']) & self.placeholders:
+                format_path['fullpath'] = paths
 
-            format_path['full'] = self.py3.composite_join(
-                format_path_separator, paths)
+            for x in self.placeholders:
+                format_path[x] = self.py3.composite_join(
+                    self.format_path_separator, format_path[x]
```

Comment:
The code here looks great. The `set()` stuffs above is not great and should be done in `post_config_hook` instead similar to `wwan.py`.

Reply:
ok